src/p4est_communication.h

```diff
@@ -575,6 +582,184 @@ void                p4est_transfer_items_end (p4est_transfer_context_t * tc);
+ * processes intersecting the point. Points known to a process before
+ * communication that do not intersect its domain are forgotten. The option
+ * \a save_unowned can be used to avoid forgetting points that do not
+ * intersect the domain of any process. If this option is enabled then these
+ * points are remembered by the process that was responsible for propagating
+ * them.
+ * 
+ * The points that a process is responsible for propagating are stored in a
+ * subarray of the array of known points, as described in 
+ * \ref p4est_transfer_search_t. Users should take care to maintain this
+ * subdivision if they modify the array of points between rounds of
+ * communication.
+ * 
+ * \param [in] p4est        The forest we search with. Its user_pointer is
+ *                          passed to the intersection callback.
+ * \param [in,out] c        Points and propagation responsibilities. The
+ *                          array \a c.points is destroyed and reallocated,
+ *                          so pointers to it should not be referenced after
+ *                          this function has been called.
+ * \param [in] intersect    Intersection callback.
+ * \param [in] save_unowned If true then points that would be unowned are
+ *                          maintained by their propagating process
+ * \return 0 if transfer was successful.
+ */
+int
+p4est_transfer_search (p4est_t *p4est, p4est_transfer_search_t *c, 
+                       p4est_intersect_t intersect, int save_unowned);
+
+/** The same as \ref p4est_transfer_search, except that we search with a
+ * partition, rather than an explicit p4est. The partition can be that of any
+ * p4est, not necessarily known to the caller.
+ * 
+ * This function is collective.
+ * 
+ * \param [in] gfq          Partition offsets to traverse.  Length \a nmemb + 1.
+ * \param [in] gfp          Partition position to traverse.  Length \a nmemb + 1.
+ * \param [in] nmemb        Number of processors encoded in \a gfq (plus one).
+ * \param [in] num_trees    Tree number must match the contents of \a gfq.
+ * \param [in] user_pointer Passed to the intersection callback.
+ * \param [in] mpicomm      Function is collective over the communicator.
+ * \param [in,out] c        Points and propagation responsibilities. The
+ *                          array \a c.points is destroyed and reallocated,
+ *                          so pointers to it should not be referenced after
+ *                          this function has been called.
+ * \param [in] intersect    Intersection callback.
+ * \param [in] save_unowned If true then points that would be unowned are
+ *                          maintained by their propagating process
+ * \return 0 if transfer was successful.
+ */
+int
+p4est_transfer_search_gfx (const p4est_gloidx_t *gfq,
+                            const p4est_quadrant_t *gfp,
+                            int nmemb, p4est_topidx_t num_trees,
```

Comment:
Thanks! I'm wondering: can nmemb be inferred by the size of the communicator, and can num_trees be read from the last element of gfp? This would save two arguments.

Reply:
On the other hand, we don't do it like this in the transfer functions. We should use the information passed for very strict assertion checking to maximize robustness.

Reply:
(This may be a duplicate comment.) Otoh, we're keeping the same redundancy in the search functions. So maybe we stick to this convention and make very sure all parameters are consistent with each other.